CONTRIBUTING.md

```diff
@@ -0,0 +1,404 @@
+* to assure the quality of the app, any PR gets reviewed, approved and tested before it will be merged to master
+
+
+### Branching model
+![branching model](/doc/branching.png "Branching Model")
+* All contributions bug fix or feature PRs target the ```master``` branch
+* Feature releases will always be based on ```master```
+* Bug fix releases will always be based on their respective feature-release-bug-fix-branches
+* Bug fixes relevant for the most recent _and_ released feature (e.g. ```11.0.0```) or bugfix (e.g. ```11.2.1```) release will be backported to the respective bugfix branch (e.g. ```stable-11.0``` or ```stable-11.2```)
+* Hot fixes not relevant for an upcoming feature release but the latest release can target the bug fix branch directly
+
+
+### Android Studio formatter setup
+Our formatter setup is rather simple:
+* Standard Android Studio
+* Line length 120 characters (Settings->Editor->Code Style->Right margin(columns): 120)
+* Auto optimize imports (Settings->Editor->Auto Import->Optimize imports on the fly)
+
+
+### Build variants
+There are three build variants
+* generic: no Google Stuff, used for F-Droid
+* gplay: with Google Stuff (Push notification), used for Google Play Store
+* qa: based on pr and available as direct download within the pr for testing purposes
+
+
+### Apply a license
+
+Nextcloud doesn't require a CLA (Contributor License Agreement). The copyright belongs to all the individual contributors. Therefore we recommend that every contributor adds following line to the header of a file, if they changed it substantially:
+
+```
+@copyright Copyright (c) <year>, <your name> (<your email address>)
+```
+
+For further information on how to add or update the license header correctly please have a look at [our licensing HowTo][applyalicense].
+
+
+### Sign your work
+
+We use the Developer Certificate of Origin (DCO) as a additional safeguard for the Nextcloud project.
+This is a well established and widely used mechanism to assure contributors have confirmed their right to license their contribution under the project's license.
+Please read [developer-certificate-of-origin][dcofile].
+If you can certify it, then just add a line to every git commit message:
+
+````
+  Signed-off-by: Random J Developer <random@developer.example.org>
+````
+
+Use your real name (sorry, no pseudonyms or anonymous contributions).
+If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
+You can also use git [aliases](https://git-scm.com/book/tr/v2/Git-Basics-Git-Aliases) like `git config --global alias.ci 'commit -s'`.
+Now you can commit with `git ci` and the commit will be signed.
+
+
+## Contribution process
+* Contribute your code in the branch ```master```. It will give us a better chance to test your code before merging it with stable code.
+* For your first contribution start a pull request on master.
```

Comment:
delete this sentence? (A PR should not only be necessary for the "first" contribution but for all contributions..)